src/foam/nanos/ruler/RulerDAO.js

```diff
@@ -127,11 +129,25 @@ if ( oldObj == null ) {
 } else {
   applyRules(x, obj, oldObj, (GroupBy) rulesList.get(getUpdateBefore()));
 }
+
+// Clone so Sync 'after' rules can be coded similar to
+// 'before' rules, whereby the rule is not responsible for 'put's and
+// subsequent rules received the updated object.
 FObject ret =  getDelegate().put_(x, obj);
-if ( oldObj == null ) {
-  applyRules(x, ret, oldObj, (GroupBy) rulesList.get(getCreateAfter()));
-} else {
-  applyRules(x, ret, oldObj, (GroupBy) rulesList.get(getUpdateAfter()));
+if ( ret != null ) {
+  ret = ret.fclone();
+  FObject before = ret.fclone();
+  if ( oldObj == null ) {
+    applyRules(x, ret, oldObj, (GroupBy) rulesList.get(getCreateAfter()));
+  } else {
+    applyRules(x, ret, oldObj, (GroupBy) rulesList.get(getUpdateAfter()));
+  }
+  // Test for changes during after rule
+  FObjectFormatter formatter = formatter_.get();
+  if ( formatter.maybeOutputDelta(before, ret) ) {
```

Comment:
You just want to know if the object has changed for journalling purposes?
If that's the case, then this would be way way faster if you just copied the maybeOutputDelta() method and kept the logic part without the outputting parts.

Reply:
Replaced with FObject.diff.